From 2f75d2abd59d89c0c571714b407d17dcbb47f9e6 Mon Sep 17 00:00:00 2001
From: Roopni Devanathan <rdevanat@qti.qualcomm.com>
Date: Mon, 8 Sep 2025 14:35:28 +0530
Subject: [PATCH] QSDK:CP:iw: Revert commit
 77de8b685e0c3d7d69a940ede4ff0ea77e1bf10b

Revert changes pertaining to setting and reading RTS threshold values for
each radio in a wiphy, because a new command for doing so in not
recommended.

Patch-dependency: 626-QSDK-CP-iw-Add-support-to-set-per-radio-RTS-threshol.patch
Patch-work: None

Signed-off-by: Roopni Devanathan <rdevanat@qti.qualcomm.com>
---
 info.c |  4 ----
 phy.c  | 41 -----------------------------------------
 2 files changed, 45 deletions(-)

diff --git a/info.c b/info.c
index 4bd089b..c9c2b73 100644
--- a/info.c
+++ b/info.c
@@ -1102,10 +1102,6 @@ broken_combination:
 				case NL80211_WIPHY_RADIO_ATTR_INDEX:
 					printf("\t\t* Idx %u:\n", nla_get_u32(radio_prop));
 					break;
-				case NL80211_WIPHY_RADIO_ATTR_RTS_THRESHOLD:
-					printf("\t\t\tRTS Threshold: %d\n",
-					       nla_get_u32(radio_prop));
-					break;
 				case NL80211_WIPHY_RADIO_ATTR_FREQ_RANGE:
 					printf("\t\t\tFrequency Range: ");
 
diff --git a/phy.c b/phy.c
index 3a01c78..f53cf56 100644
--- a/phy.c
+++ b/phy.c
@@ -668,47 +668,6 @@ COMMAND(set, rts, "[-l] <link_id> <rts threshold|off>",
 	NL80211_CMD_SET_WIPHY, 0, CIB_NETDEV, handle_rts,
 	"Set rts threshold.");
 
-static int handle_wiphy_params(struct nl80211_state *state,
-			       struct nl_msg *msg,
-			       int argc, char **argv,
-			       enum id_input id)
-{
-	unsigned int rts = -1, radio_idx = NL80211_WIPHY_RADIO_ID_MAX;
-	int argidx = 0;
-	char *end;
-
-	if (argc < 2)
-		return 1;
-
-	if (!strcmp("radio", argv[argidx])) {
-		radio_idx = strtoul(argv[argidx + 1], &end, 10);
-		if (*end != '\0')
-			return 1;
-
-		argidx = argidx + 2;
-	}
-
-	NLA_PUT_U8(msg, NL80211_ATTR_WIPHY_RADIO_INDEX, radio_idx);
-
-	if (!strcmp("rts", argv[argidx]) && (strcmp("off", argv[argidx + 1])))
-		rts = strtoul(argv[argidx + 1], &end, 10);
-	else
-		rts = -1;
-
-	if (*end != '\0')
-		return 1;
-
-	NLA_PUT_U32(msg, NL80211_ATTR_WIPHY_RTS_THRESHOLD, rts);
-
-	return 0;
-nla_put_failure:
-	return 1;
-}
-
-COMMAND(set, wiphy, "[radio <idx>] rts <rts_threshold value>",
-	NL80211_CMD_SET_WIPHY, 0, CIB_PHY, handle_wiphy_params,
-	"Set per-radio wiphy parameters");
-
 static int handle_retry(struct nl80211_state *state,
 			struct nl_msg *msg,
 			int argc, char **argv, enum id_input id)
-- 
2.34.1

